common/constants.go

```diff
@@ -264,3 +264,8 @@ const (
 	// DefaultHistoryMaxAutoResetPoints is the default maximum number for auto reset points
 	DefaultHistoryMaxAutoResetPoints = 20
 )
+
+const (
+	DeprecatedDomainStatus = 1
+	DeletedDomainStatus    = 2
+)
```

Comment:
these are still copies of original enum. see my prev comment https://github.com/uber/cadence/pull/5580#discussion_r1443319705

Reply:
@Shaddoll what is the pattern we follow for defining a corresponding type in common that is originally defined in persistence package only?

Reply:
I tried importing the original enum but it didn't work out so ended up defining these in a common file.

Reply:
@agautam478 why didn't it work when imported from persistence? cyclic dependency?
We need to figure out a solution than re-defining these because there's no mechanism ensuring these don't go out of sync